virttest/qemu_devices/qdevices.py

```diff
@@ -459,6 +475,20 @@ def cmdline_nd(self):
             out = out[:-1]
         return out
 
+    @staticmethod
+    def _convert_raw_style_to_json_style_args(args):
+        """
+        Convert string type of 'on' and 'off' to boolean, and create new dict
+        (e.g: 'cache': {'direct': 'true'}) from key which include symbol '.'
+        (e.g: 'cache.direct': 'true') to adhere to the blockdev qmp syntax.
+
+        :param args: Dictionary with the qmp parameters.
+        :type args: dict
+        :return: Converted args.
+        :rtype: dict
+        """
+        return NotImplementedError
```

Comment:
@nickzhq One quick comment, I think here should not be `NotImplementedError`, I suppose we should implement json format at the same level with raw format. We have to find out the regular pattern of json format among same type command line(so here in virttest they are in same QxxDevice class), and then implement them at a common level. If the current params structure can not fit the new format pattern, we need to twist params structure instead of implement  command line generator separately.
We could discuss it offline.